### PR TITLE
add new directories and files

### PR DESCRIPTION
**Pull request by Caren-Koroeny**
<h1> Hello Code Reviewer</h1>

> In this PR I did the following tasks as required:

- `redux toolkit`  is present in package.json 

       "dependencies": {
        "@reduxjs/toolkit": "^1.9.3",

- The structure of the files should followed the `ducks` `pattern.`
     
Screenshot from 2023-03-06 20-24-53

- Add Redux Toolkit `(npm install react-redux @reduxjs/toolkit)`.
 
- Create a directory that will contain all your Redux logic `(/src/redux)`.

- Configure a Redux store `(/src/redux/store.js)`.

- Defined  a `slice` of state for `books` that: it is found in `bookSlice.js` file

      - Includes an array of books (initial state: empty array)

      - Includes a reducer that adds a book.

    -  Includes a reducer that removes a book.

- Define a slice of state for categories that: it is found in  `categorieSlice.js`

       - Includes an array of categories (initial state: empty array)

       - Includes a reducer that check the status and always return "Under construction"

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  type: ADD_BOOK,
+  book,
+});
+
+const RemoveBook = (bookId) => ({
+  type: REMOVE_BOOK,
+  bookId,
+});
+
+export { BookReducer, AddBook, RemoveBook };
```
Comment (muneebulrehman) [inline src/redux/books/bookSlice.js]: Kindly use the `createSlice` function to automatically generate action creators and action types that will correspond to the reducers and state. I see you have used `createSlice` in categoriesSlice.js`. I recommend you follow the same approach here. Remember to export the reducer actions once you create them.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Caren-Koroeny , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

- [ ] I would recommend removing the Microverse badge that you have included in the README 
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/bookSlice.js (hunk 1 above)
Comment (Caren-Koroeny) [reply src/redux/books/bookSlice.js]: Okay, I fixed it, thank you for the corrections.

Screenshot from 2023-03-07 08-23-12

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**